lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -1086,3 +1115,84 @@ def _zalpha(colors, zs):
     sats = 1 - norm(zs) * 0.7
     rgba = np.broadcast_to(mcolors.to_rgba_array(colors), (len(zs), 4))
     return np.column_stack([rgba[:, :3], rgba[:, 3] * sats])
+
+
+def _generate_normals(polygons):
+    """
+    Compute the normals of a list of polygons.
+
+    Normals point towards the viewer for a face with its vertices in
+    counterclockwise order, following the right hand rule.
+
+    Uses three points equally spaced around the polygon.
```

Comment:
I have not modified this doc-string as the functions are just methods moved from Axes3D.

However, I have looked into it it a bit earlier in another issue. A polygon should be in a plane (if it is not, strange things are likely to happen). And a plane is well defined by three points, so basically any three points can be used. I guess that "equally spaced" is to avoid numerical issues if the points are close together (which statistically should be better separated).